src/test/java/env/DriverUtil.java

```diff
@@ -113,35 +107,61 @@ private static WebDriver iosDriver(DesiredCapabilities capabilities) throws Malf
+        capabilities = setupDefaultDesktopCapabilities(DesiredCapabilities.edge());
+        driver = new EdgeDriver(capabilities);
+        return driver;
+    }
+
+    private static WebDriver firefoxDriver(boolean headless) {
+        DesiredCapabilities capabilities;
+        capabilities = setupDefaultDesktopCapabilities(DesiredCapabilities.firefox());
+        final FirefoxOptions options = new FirefoxOptions();
+        if (headless) {
+            options.addArguments("-headless", "-safe-mode");
+        }
+        capabilities.setCapability(FirefoxOptions.FIREFOX_OPTIONS, options);
+        driver = new FirefoxDriver(capabilities);
+        return driver;
+    }
+
+    private static WebDriver chromeDriver(boolean headless) {
+        DesiredCapabilities capabilities;
+        capabilities = setupDefaultDesktopCapabilities(DesiredCapabilities.chrome());
+        final ChromeOptions chromeOptions = new ChromeOptions();
+        if (headless) {
+            chromeOptions.addArguments("--headless");
+        }
+        capabilities.setCapability(ChromeOptions.CAPABILITY, chromeOptions);
+        driver = new ChromeDriver(capabilities);
+        ErrorHandler handler = new ErrorHandler();
+        handler.setIncludeServerErrors(false);
+        return driver;
+    }
+
+    private static WebDriver chooseDriver() {
         String preferredDriver = System.getProperty("browser", "Firefox");
         boolean headless = System.getProperty("headless", "false").equals("true");
 
         switch (preferredDriver.toLowerCase()) {
             case "safari":
-                driver = new SafariDriver();
-                return driver;
+                return safariDriver();
             case "edge":
-                driver = new EdgeDriver();
-                return driver;
+                return edgeDriver();
             case "chrome":
-                final ChromeOptions chromeOptions = new ChromeOptions();
-                if (headless) {
-                    chromeOptions.addArguments("--headless");
-                }
-                capabilities.setCapability(ChromeOptions.CAPABILITY, chromeOptions);
-                driver = new ChromeDriver(capabilities);
-                ErrorHandler handler = new ErrorHandler();
-                handler.setIncludeServerErrors(false);
-                return driver;
+                return chromeDriver(headless);
+            case "firefox":
+                return firefoxDriver(headless);
             default:
-                FirefoxOptions options = new FirefoxOptions();
-                if (headless) {
-                    options.addArguments("-headless", "-safe-mode");
-                }
-                capabilities.setCapability(FirefoxOptions.FIREFOX_OPTIONS, options);
-                driver = new FirefoxDriver(capabilities);
-                return driver;
+                throw new RuntimeException(String.format("Not supported browser: %s", preferredDriver));
```

Comment:
Use IllegalArgumentException, RuntimeException is too generic and should be avoided